inc/classes/subscriber/third-party/plugins/class-smush-subscriber.php

```diff
@@ -37,34 +44,88 @@ public static function get_subscribed_events() {
 	 * @author Soponar Cristina
 	 */
 	public function maybe_deactivate_rocket_lazyload() {
-		$lazy_load_option = get_option( rocket_get_constant( 'WP_SMUSH_PREFIX' ) . 'settings' );
-		$lazyload         = isset( $lazy_load_option['lazy_load'] ) ? $lazy_load_option['lazy_load'] : false;
+		$enabled = $this->is_smush_lazyload_enabled();
 
-		if ( ! $lazyload ) {
-			return;
+		if ( $enabled['images'] && get_rocket_option( 'lazyload' ) ) {
+			update_rocket_option( 'lazyload', 0 );
 		}
 
-		update_rocket_option( 'lazyload', 0 );
+		if ( $enabled['iframes'] && get_rocket_option( 'lazyload_iframes' ) ) {
+			update_rocket_option( 'lazyload_iframes', 0 );
+		}
 	}
 
 	/**
-	 * Disable WP Rocket lazyload fields if WP Smush lazyload is enabled
+	 * Disable WP Rocket lazyload field for images if WP Smush lazyload is enabled.
 	 *
 	 * @since  3.4.2
 	 * @author Soponar Cristina
 	 *
-	 * @param array $rocket_maybe_disable_lazyload_plugins Array with plugins which disable lazyload functionality.
-	 * @return string
+	 * @param  array $disable_images_lazyload Array with plugins which disable lazyload functionality.
+	 * @return array                          A list of plugin names.
+	 */
+	public function is_smush_lazyload_active( $disable_images_lazyload ) {
+		$enabled = $this->is_smush_lazyload_enabled();
+
+		if ( $enabled['images'] ) {
+			$disable_images_lazyload[] = __( 'Smush', 'rocket' );
+		}
+
+		return $disable_images_lazyload;
+	}
+
+	/**
+	 * Tell if Smush’s lazyload is enabled for each type of content.
+	 *
+	 * @since  3.5.1
+	 * @author Grégory Viguier
+	 *
+	 * @return array {
+	 *     @var bool $images  True when lazyload is enabled for images. False otherwise.
+	 *     @var bool $iframes True when lazyload is enabled for iframes. False otherwise.
+	 * }
 	 */
-	public function is_smush_lazyload_active( $rocket_maybe_disable_lazyload_plugins ) {
-		$lazy_load_option = get_option( rocket_get_constant( 'WP_SMUSH_PREFIX' ) . 'settings' );
-		$lazyload         = isset( $lazy_load_option['lazy_load'] ) ? $lazy_load_option['lazy_load'] : false;
+	private function is_smush_lazyload_enabled() {
+		$enabled = [
+			'images'  => false,
+			'iframes' => false,
+		];
+
+		if ( ! class_exists( '\\Smush\\Core\\Settings' ) ) {
+			return $enabled;
+		}
+
+		$settings = Settings::get_instance();
+
+		if ( ! $settings->get( 'lazy_load' ) ) {
+			return $enabled;
+		}
+
+		$prefix  = rocket_get_constant( 'WP_SMUSH_PREFIX', 'wp-smush-' );
+		$formats = $settings->get_setting( $prefix . 'lazy_load' );
+		$formats = ! empty( $formats['format'] ) && is_array( $formats['format'] ) ? array_filter( $formats['format'] ) : [];
+
+		$image_formats = array_intersect_key(
+			$formats,
+			// Whitelist image formats.
+			[
+				'jpeg' => false,
+				'png'  => false,
+				'gif'  => false,
+				'svg'  => false,
+			]
+		);
+
+		if ( $image_formats ) {
```

Comment:
This is an array. Let's not treat it like a boolean. Check if it's not empty for the conditional.